### PR TITLE
feat: port to the app-platform

### PR DESCRIPTION
**Pull request by HendrikThePendric**
# DO NOT MERGE BEFORE THE FREEZE

This PR ports the user-app to the app-platform. As you can see, some checks are still failing. This is due to the fact that the cli-style checks are producing errors. I have fixed these errors in a separate branch because I wanted to not introduce too much clutter into this one.
My recommendation would be to follow these steps:
1. Thoroughly review the current PR, which doesn't include all this clutter right now.
2. Briefly review the PR #580 which is 99% just changing the propTypes-order and removing/adding some propTypes.
3. Merge PR #580 into the current PR and if the checks pass, it can be approved

### CODE REVIEW

File: d2.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const config = {
```
Comment (ismay) [inline d2.config.js:1]: Maybe we should add a description as well: https://platform.dhis2.nu/#/config/d2-config-js-reference (I'm not sure if that's a requirement).

File: src/App.js [hunk 2]
```diff
@@ -1,54 +1,28 @@
...
-import { HeaderBar } from '@dhis2/ui-widgets'
-import { CssReset } from '@dhis2/ui-core'
-
-import i18n from '@dhis2/d2-i18n'
-
-import 'typeface-roboto'
-import 'material-design-icons-iconfont/dist/material-design-icons.css'
-
 /**
  * Main Component. Renders a AppWithD2ContextAndTheme wrapped in a Provider containing the HeaderBar,
```
Comment (ismay) [inline src/App.js:12]: This is no longer true right?

File: src/AppWrapper.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            ],
+        },
+        i18nRoot: './d2i18n',
+        onInitialized: d2 => {
+            api.init(d2)
+        },
+    })
+
+    if (!d2) {
+        return null
```
Comment (ismay) [inline src/AppWrapper.js:31]: Maybe it'd be good to return a centered loading spinner here.

**Comment by ismay** [comment]
Looks good to me. Left a couple of comments on things that weren't clear to me and we should address the failing CI tasks (I assume that that's the separate PR), but other than that it looks good to merge to me.

Unrelated issue, I see that there's no way to go back to the 'home' route via the menu on the left. I'd expect there to be a home/overview menu item.

File: .gitignore [hunk 4]
```diff
@@ -17,11 +15,15 @@
...
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 
 # generated files
 src/locales
 docs
+
+# DHIS2 Platform
+node_modules
```
Comment (Mohammer5) [inline .gitignore:27]: duplicate, see line 2

**Review by Mohammer5 — APPROVED** [review]
Apart from @ismay's comments, this looks really good to me :+1:

File: src/App.js (hunk 2 above)
Comment (HendrikThePendric) [reply src/App.js]: Good find, I updated the comment. I wish I would have never added these jsDocs to this app. They just get out of sync with the actual code so easily 😞

File: d2.config.js (hunk 1 above)
Comment (HendrikThePendric) [reply d2.config.js:1]: It's not required, but I added one anyway.

**Review by ismay — APPROVED** [review]

**Pull request merged**